### PR TITLE
Add support for listing metadata

### PR DESCRIPTION
**Pull request by kirre-bylund**

### CODE REVIEW

**Review by JohannesLoot — APPROVED** [review]
Coudln't find anything at all to say anything about.
I have nothing to say other than awesome work, it's so well documented, structured and even with extra QOL functions, beautiful work ⭐

**Review by AlmightyMikkel — APPROVED** [review]
Once again, really great work with the implementation of Metadata!

File: LootLockerServerSDK/Source/LootLockerServerSDK/Private/ServerAPI/LootLockerServerMetadataRequest.cpp [hunk 1]
```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2021 LootLocker
+
+#include "ServerAPI/LootLockerServerMetadataRequest.h"
+
+#include "LootLockerServerHttpClient.h"
+#include "Dom/JsonObject.h"
```
Comment (kirre-bylund) [inline LootLockerServerSDK/Source/LootLockerServerSDK/Private/ServerAPI/LootLockerServerMetadataRequest.cpp:6]: Is this good?

File: LootLockerServerSDK/Source/LootLockerServerSDK/Public/ServerAPI/LootLockerServerMetadataRequest.h [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    LOOTLOCKERSERVERSDK_API bool TryGetValueAsInteger(int& Output) const;
+    /*
+     Get the value as a boolean. Returns true if value could be parsed in which case Output contains the bool, returns false if parsing failed which can happen if the string is not a convertible to a boolean (those are for example "0", "1", "true", "False", "yes", "NO", etc).
+     */
+    LOOTLOCKERSERVERSDK_API bool TryGetValueAsBool(bool& Output) const;
+    /*
+     Get the value as an unparsed json value. Returns true if value could be found in which case Output contains the JsonValue, returns false if the value field was not present.
+     */
+    LOOTLOCKERSERVERSDK_API bool TryGetRawValue(TSharedPtr<FJsonValue>& Output) const;
+    LOOTLOCKERSERVERSDK_API bool TryGetSerializedValue(FString& Output) const;
```
Comment (kirre-bylund) [inline LootLockerServerSDK/Source/LootLockerServerSDK/Public/ServerAPI/LootLockerServerMetadataRequest.h:121]: Missing documentation

File: LootLockerServerSDK/Source/LootLockerServerSDK/Private/ServerAPI/LootLockerServerMetadataRequest.cpp (hunk 1 above)
Comment (kirre-bylund) [reply LootLockerServerSDK/Source/LootLockerServerSDK/Private/ServerAPI/LootLockerServerMetadataRequest.cpp:6]: It is the Unreal Json object so it's fine

**Pull request merged**